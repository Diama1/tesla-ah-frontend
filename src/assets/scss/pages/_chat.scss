.chat{
  &__container{
    min-height: 80vh;
  }
  &__box{
    margin:auto 20%;
    border:1px solid lighten($grey,50%);
    padding: 20px;
    border-radius: 5px;
    overflow:scroll;
    height: 100vh;
  }
  &__message{
    border:1px solid $grey;
    padding: 8px;
    border-radius: 15px;
    margin:10px auto;
    display: flex;
    justify-content: space-between;
    width: 50%;
    margin-right: 50%;
    margin-left: 0;
    @include mediaSm{
      display: block;
    }
    
  }
  &__mine{
    background:$grey-ish;
    color:$white;
    margin-left:50%;
    border:0;
    p{
      margin-right:20px;
    }
  }
  &__avatar{
    width: 40px;
    height: 40px;
    border-radius: 50px;
    border: 1px solid $grey;
    margin-top:10px;
    @include mediaSm{
      width: 20px;
      height: 20px;
      border-radius: 50%;
    }
  }
  &__form{
    display:flex;
    justify-content: space-around;
  }
  &__header{
    display:flex;
    margin: auto 20%;
    i{
      font-size:40px;
    }
  }
  &__friend{
    margin-left: 30px;
  }
  &__message{
    &--data{
      margin: auto 10px;
    }
  }
  &__info{
    position:relative;
    i{
      position: relative;
      color:$grey;
    }
  }
  &__user{
    display: flex;
    justify-content: space-between;
    &--name{
      margin-top:15px;
      margin-right:20px;
      margin-left:10px;
      p{
        color:$dark;
      }
    }
    &--status{
      margin-top:15px;
      i{
        color:$grey;
      }
    }
    &--info{
      display: flex;
      justify-content: space-evenly;
    }
  }

  &__back{
    color:$dark;
  }
  &__username{
    color:$dark;
  }
}

.users{
  &__container{
    margin:auto 20%;
  }
  &__title{
    margin-bottom: 5px;
    margin-left:15%;
  }
}