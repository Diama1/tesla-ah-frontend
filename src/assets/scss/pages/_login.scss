@import '../utils/mobile';
.main-section-login {
  .row-login {
    min-height: 400px;
    margin: 0 auto;
    width: 60vh;
    margin-top: 100px;
    border-radius: 7px;
    padding: 70px 0 70px 0;
    box-shadow: 2px 2px 10px 2px #c9c9c9;
    @include mediaSm {
      width: 100%;
    }
    @include mediaMd {
      width: 100%;
      margin-top: 100px;
      box-shadow: 0px 0px 0px 0px #ffffff;
    }

    #col-login {
      width: 100%;
      margin: 0 auto;

      h4 {
        color: $title_color;
        font-weight: 500;
        text-shadow: 0px 4px 4px rgba(0, 174, 255, 0.25);
        letter-spacing: 0.02em;
        text-align: center;
      }
    }
  }
}
