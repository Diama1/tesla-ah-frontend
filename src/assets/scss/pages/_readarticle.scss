.body-content{
    margin: 20%;
}