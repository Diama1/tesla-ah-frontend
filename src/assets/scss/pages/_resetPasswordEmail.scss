@import '../utils/variables';
@import '../layouts/resetPasswordEmailForm';


main{
  margin:auto 0%;
  transform:translate3d(0,-20px,0);
  .rowForm {
    min-height: 400px;
    margin: 0 auto;
    width: 80%;
    margin-top: 6%;
    border-radius: 7px;
    padding:70px 3% ;
    box-shadow: 2px 2px 10px 2px #C9C9C9;
    .col-md-6{
      h4{
        color: $title_color;
        font-weight: 500;
        text-shadow:  0px 4px 4px rgba(0, 174, 255, 0.25);
        letter-spacing: 0.02em;

      }
       h5{
        color: $black;
        margin-top: 3%;
        font-weight: 450;
        letter-spacing: 0.02em;

      }
      .reset-message{
        color: rgb(66, 66, 136)
      }
      .link{
        font-size: 22px;
        margin-left: 3%;
      }
    }
    #passwordImage{
      margin-top: 4%;
      img{
        width:400px;
      }
    }
  }
}
.main-section{
  margin: auto 11.5%;
}

.side-section{
  border-left: 1px solid rgb(226, 226, 226);
  padding-left: 2.1vh;
}
@media(max-width: 768px){
  
  main .row #image
  {
    margin-top: 300px;

  img{
      width: 230px
    }
  } 
 main .row #passwordImage
 {
  display: none;
} 
}
@media(max-width: 480px){
  #image,#passwordImage{
    display: none;
  }
  .col-md-6{
    
    h4{
      color:$title_color;
      font-weight: 500;
      text-align: center;
      text-shadow:  0px 4px 4px rgba(0, 174, 255, 0.25);
      letter-spacing: 0.02em;

    }
     h5{
        color: $black;
        margin-top: 3%;
        font-weight: 450;
        text-align: center;
        letter-spacing: 0.02em;

      }
  }
  
}
@media(max-width: 1024px){

  main .row #passwordImage img{
    width: 250px
  }
}
@media(width: 1024px){

  main .row #image img{
    width: 300px
  }
}