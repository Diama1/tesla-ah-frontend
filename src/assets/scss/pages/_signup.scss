.signup {
  &__section {
    border: 1px solid $grey;
    border-radius: 10px;
    padding: 20px;
  }

  &__people{
    margin-top: 20%;
  }
}

.verify{
  &__container{
    min-height: 70vh;
  }
}
.alert{
  &__container{
    margin-top:20px;
  }
}
