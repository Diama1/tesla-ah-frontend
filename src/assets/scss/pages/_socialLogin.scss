.carousel-login{
    border: 1px solid black;
    height: 70vh;
   & .carousel-login-div{
     margin-top: 15vh;
     margin-left: 20vh;
     width: 40vh;
     height: 40vh;
   }
  }
  #login-dp{
    min-width: 250px;
    padding: 14px 14px 0;
    overflow:hidden;
    background-color:rgba(255,255,255,.8);
  }
  #login-dp .help-block{
    font-size:12px    
  }
  #login-dp .bottom{
    background-color:rgba(255,255,255,.8);
    border-top:1px solid #ddd;
    clear:both;
    padding:14px;
  }
  #login-dp .social-buttons{
    margin:12px 0    
  }
  #login-dp .social-buttons a{
    width: 49%;
  }
  #login-dp .form-group {
    margin-bottom: 10px;
  }
  .btn-fb{
    color: $white;
    background-color:#3b5998;
  }
  .checkbox{
    color: $white;
  }
  .btn-fb:hover{
    color: $white;
    background-color:#496ebc 
  }
  .btn-tw{
  color: $white;
  background-color:#55acee;
  }
  .btn-tw:hover{
  color: $white;
  background-color:#59b5fa;
  }
  @media(max-width:768px){
    #login-dp{
    background-color: inherit;
    color: $white;
    }
    #login-dp .bottom{
    background-color: inherit;
    border-top:0 none;
    }
  }
@media(max-width:480px){
     .carousel-login{& .carousel-login-div{
      height: 40vh;
      margin-left: 4vh;
      margin-top: 10vh;
      width: 85%;    
     }
    }
  }
