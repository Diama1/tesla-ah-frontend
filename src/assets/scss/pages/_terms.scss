.terms {
  padding: 30px;
  position: relative;
  &__pub {
    &--1 {
      display: block;
      margin-bottom: 10px;
    }

  }

}
