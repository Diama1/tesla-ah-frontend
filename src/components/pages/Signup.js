/* eslint-disable react/no-unused-state */
/* eslint-disable jsx-a11y/label-has-for */
/* eslint-disable jsx-a11y/label-has-associated-control */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { toast } from 'react-toastify';
import PropTypes from 'prop-types';
import happyFaces from '../../assets/img/signup.png';
import Spinner from '../widgets/Spinner';
import Alert from '../common/Alert';
import { createAccount } from '../../redux/actions/auth.actions';
import validate from '../../utils/validations';


export class Signup extends Component {
  constructor(props) {
    super(props);
    this.state = {
      firstName: '',
      lastName: '',
      email: '',
      username: '',
      password: '',
      confirmPassword: '',
      gender: 'M',
      bio: '',
      birthDate: '',
    };
  }

  componentDidMount() {
  }

  handleSubmit = (e) => {
    e.preventDefault();

    const formErrors = validate(this.state, 'signup');
    if (Object.values(formErrors).length) {
      Object.values(formErrors).forEach((err) => {
        toast.error(err);
      });
    } else {
      const {
        state: {
          firstName,
          lastName,
          email,
          username,
          password,
          confirmPassword,
          bio,
          birthDate,
        },
      } = this;
      // eslint-disable-next-line react/destructuring-assignment
      this.props.createAccount({
        firstName,
        lastName,
        email,
        username,
        password,
        confirmPassword,
        bio,
        dateOfBirth: birthDate,
      });
    }
  }

  render() {
    const {
      firstName, lastName, email, username, password, confirmPassword, bio, birthDate,
    } = this.state;
    const { ui: { loading }, auth: { signupSuccess } } = this.props;
    return (
      <section className="container">
        <main className="row mt-5 mb-5 signup__section">
          <div className="col-md-6 pt-5 pb-5">
            <h4 className="mb-5">Join Authors Haven</h4>
            {signupSuccess ? (
              <Alert type="success">
                <p>
Thank you for creating an account with Authors Haven,
                you will receive an email shortly!

                </p>
              </Alert>
            ) : (
              <form onSubmit={this.handleSubmit}>
                <div className="row">
                  <div className="col-sm-6">
                    <div className="form-group">
                      <input
                        type="text"
                        name="firstName"
                        id="firstName"
                        className="form-control"
                        placeholder="First Name"
                        value={firstName}
                        onChange={e => this.setState({ firstName: e.target.value })}
                        required
                      />
                    </div>
                  </div>
                  <div className="col-sm-6">
                    <div className="form-group">
                      <input
                        type="text"
                        name="lastName"
                        id="lastName"
                        className="form-control"
                        placeholder="Last Name"
                        value={lastName}
                        onChange={e => this.setState({ lastName: e.target.value })}
                        required
                      />
                    </div>
                  </div>
                </div>
                <div className="form-group">
                  <input
                    type="email"
                    name="email"
                    placeholder="Email Address"
                    className="form-control"
                    value={email}
                    onChange={e => this.setState({ email: e.target.value })}
                    required
                  />
                </div>
                <div className="form-froup">
                  <input
                    type="text"
                    className="form-control"
                    placeholder="Username"
                    name="username"
                    value={username}
                    onChange={e => this.setState({ username: e.target.value })}
                    required
                  />
                </div>
                <div className="row form-group mt-3">
                  <div className="col-sm-6">
                    <div className="form-group">
                      <input
                        type="password"
                        className="form-control"
                        placeholder="Password"
                        name="password"
                        value={password}
                        onChange={e => this.setState({ password: e.target.value })}
                        required
                      />
                    </div>
                  </div>
                  <div className="col-sm-6">
                    <div className="form-group">
                      <input
                        type="password"
                        className="form-control"
                        placeholder="Confirm"
                        name="confirm"
                        value={confirmPassword}
                        onChange={e => this.setState({ confirmPassword: e.target.value })}
                        required
                      />
                    </div>
                  </div>
                </div>
                <div className="form-group">
                  <label htmlFor="birthdate">Birth date</label>
                  <input
                    type="date"
                    className="form-control"
                    placeholder="Birth Date"
                    value={birthDate}
                    name="birthDate"
                    onChange={e => this.setState({ birthDate: e.target.value })}
                    required
                  />
                </div>
                <div className="form-group"><label htmlFor="gender">Gender</label></div>
                <div className="row form-radios">
                  <div className="form-check col-sm-6 form-group">
                    <input
                      type="radio"
                      className="form-check-input"
                      name="gender"
                      value="M"
                      onChange={e => this.setState({ gender: e.target.value })}
                      checked
                    />
                    <label className="form-check-label">Male</label>
                  </div>
                  <div className="form-check col-sm-6">
                    <input
                      type="radio"
                      className="form-check-input"
                      name="gender"
                      value="F"
                      onChange={e => this.setState({ gender: e.target.value })}
                    />
                    <label className="form-check-label">Female</label>
                  </div>
                </div>
                <div className="form-group">
                  <textarea
                    placeholder="Bio"
                    value={bio}
                    className="form-control"
                    onChange={e => this.setState({ bio: e.target.value })}
                  />
                </div>
                <div className="form-check mb-4">
                  {/* eslint-disable-nextline  */}
                  <Link to="/terms">Read terms and conditions first...</Link>
                  <br />
                  <input type="checkbox" className="form-check-input" required />
                  <label htmlFor="Checkbox" className="form-check-label">I agree with the terms and conditions...</label>
                </div>
                <button type="submit" className="btn button is-grey">Register</button>
                {loading ? (
                  <Spinner caption="Registering..." />
                ) : null}
                <div className="mt-3">
                  <p>
                      Already have an account?
                    {' '}
                    {/* eslint-disable-nextline  */}
                    <Link to="/auth/login">Login</Link>
                  </p>
                </div>
              </form>
            )}

          </div>
          <div className="col-md-6">
            <img src={happyFaces} alt="AH People" className="signup__people" />
          </div>
        </main>
      </section>
    );
  }
}

const mapStateToProps = state => ({
  ui: state.ui,
  auth: state.auth,
});

Signup.propTypes = {
  createAccount: PropTypes.func.isRequired,
  ui: PropTypes.instanceOf(Object).isRequired,
  auth: PropTypes.instanceOf(Object).isRequired,
};

export default connect(mapStateToProps, { createAccount })(Signup);
