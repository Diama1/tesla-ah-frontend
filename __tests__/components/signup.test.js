/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import { Signup } from '../../src/components/pages/Signup';

const props = {
  createAccount: jest.fn(),
  ui: { loading: false },
  auth: { signupSuccess: false },
};

const otherProps = {
  ui: { loading: true },
  createAccount: jest.fn(),
  auth: { signupSuccess: false },
};
let signup;
let signupLoading;
describe('Signup Components tests...', () => {
  beforeAll(() => {
    signup = shallow(<Signup {...props} />);
  });

  it('Should render signup component', () => {
    expect(signup).toMatchSnapshot();
  });
  it('Should find one form', () => {
    const form = signup.find('form');
    expect(form.length).toBe(1);
  });
  it('Should give initial state', () => {
    expect(signup.state()).toBeDefined();
  });
  describe('Input simulations tests...', () => {
    it('Type in firstName field', () => {
      const firstName = signup.find('input[name="firstName"]');
      firstName.simulate('change', {
        target: { value: 'Elie' },
      });
      expect(signup.state('firstName')).toEqual('Elie');
    });
    it('Type in lastName field', () => {
      const lastName = signup.find('input[name="lastName"]');
      lastName.simulate('change', {
        target: { value: 'Mugenzi' },
      });
      expect(signup.state('lastName')).toEqual('Mugenzi');
    });
    it('Change the state on email typing', () => {
      const email = signup.find('input[name="email"]');
      email.simulate('change', {
        target: { value: 'elie@me.com' },
      });
      expect(signup.state('email')).toEqual('elie@me.com');
    });
    it('Change the username state on type', () => {
      const username = signup.find('input[name="username"]');
      username.simulate('change', {
        target: { value: 'elie' },
      });
      expect(signup.state('username')).toEqual('elie');
    });
    it('Should type the password and update its state', () => {
      const password = signup.find('input[name="password"]');
      password.simulate('change', {
        target: { value: '123456' },
      });
      expect(signup.state('password')).toEqual('123456');
    });
    it('Should change confirmPassword state', () => {
      const confirm = signup.find('input[name="confirm"]');
      confirm.simulate('change', {
        target: { value: '123456' },
      });
      expect(signup.state('confirmPassword')).toEqual('123456');
    });
    it('Should change DOB state', () => {
      const dob = signup.find('input[name="birthDate"]');
      dob.simulate('change', {
        target: { value: '01/03/1997' },
      });
      expect(signup.state('birthDate')).toEqual('01/03/1997');
    });
    it('Should update Gender to M', () => {
      const gender = signup.find('input[name="gender"]');
      gender.first().simulate('change', {
        target: { value: 'M' },
      });
      expect(signup.state('gender')).toEqual('M');
    });
    it('Should update gender to F', () => {
      const gender = signup.find('input[name="gender"]');
      gender.at(1).simulate('change', {
        target: { value: 'F' },
      });
      expect(signup.state('gender')).toEqual('F');
    });
    it('Should update BIO state', () => {
      const bio = signup.find('textarea');
      bio.simulate('change', {
        target: { value: 'Human being' },
      });
      expect(signup.state('bio')).toEqual('Human being');
    });
  });

  describe('Simulate form submission', () => {
    let instance;
    let submitButton;
    beforeAll(() => {
      instance = signup.instance();
      submitButton = signup.find('button[type="submit"]');
      submitButton.simulate('click');
    });
    it('Should make a remote request to the server', () => {
      instance.forceUpdate();
      signup.update();
      const event = {
        preventDefault: jest.fn(),
      };
      instance.handleSubmit(event);
    });
  });

  describe('Loading now', () => {
    beforeAll(() => {
      signupLoading = shallow(<Signup {...otherProps} />);
    });
    it('Renders successfully', () => {
      expect(signupLoading).toBeDefined();
    });
  });

  describe('On signup success', () => {
    let signUpDone;
    const thoseProps = {
      ui: { loading: false },
      createAccount: jest.fn(),
      auth: { signupSuccess: true },
    };
    beforeAll(() => {
      signUpDone = shallow(<Signup {...thoseProps} />);
    });
    it('Renders successfully', () => {
      expect(signUpDone).toBeDefined();
    });
  });
});
