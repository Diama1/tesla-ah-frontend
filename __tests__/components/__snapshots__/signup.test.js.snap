// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup Components tests... Should render signup component 1`] = `ShallowWrapper {}`;

exports[`Signup Components tests... Should render signup component 2`] = `ShallowWrapper {}`;
